src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -819,32 +792,233 @@ bool AutomationInstrumentClipView::renderSidebar(uint32_t whichRows, uint8_t ima
+			}
+		}
+		else if (instrument->type != InstrumentType::CV) {
+			/* check if you're holding a pad
+			* if yes, store pad press knob position in lastPadSelectedKnobPos
+			* so that it can be used next time as the knob position if returning here
+			* to display parameter value after another popup has been cancelled (e.g. audition pad)
+			*/
+			if (isUIModeActive(UI_MODE_NOTES_PRESSED)) {
+				if (knobPosLeft != kNoSelection) {
+					lastPadSelectedKnobPos = knobPosLeft;
+				}
+				else if (lastPadSelectedKnobPos != kNoSelection) {
+					knobPosLeft = lastPadSelectedKnobPos;
 				}
 			}
+
+			//display parameter value if knobPos is provided
+			if (knobPosLeft != kNoSelection) {
+				char buffer[5];
+
+				intToString(knobPosLeft, buffer);
+
+				display->setText(buffer, false, 255, false);
+			}
+			//display parameter name
+			else {
+				char parameterName[30];
+				getParameterName(parameterName);
+				display->setScrollingText(parameterName);
+			}
+		}
+	}
+}
+
+//get's the name of the Parameter being edited so it can be displayed on the screen
+void AutomationInstrumentClipView::getParameterName(char* parameterName) {
+
+	InstrumentClip* clip = getCurrentClip();
+	Instrument* instrument = (Instrument*)clip->output;
+
+	if (instrument->type == InstrumentType::SYNTH || instrument->type == InstrumentType::KIT) {
+		if (clip->lastSelectedParamKind == Param::Kind::PATCHED) {
+			strncpy(parameterName, getPatchedParamDisplayName(clip->lastSelectedParamID), 29);
+		}
+		else if (clip->lastSelectedParamKind == Param::Kind::UNPATCHED) {
+			strncpy(parameterName, getUnpatchedParamDisplayName(clip->lastSelectedParamID), 29);
+		}
+		else if (clip->lastSelectedParamKind == Param::Kind::GLOBAL_EFFECTABLE) {
+			strncpy(parameterName, getGlobalEffectableParamDisplayName(clip->lastSelectedParamID), 29);
+		}
+	}
+	else if (instrument->type == InstrumentType::MIDI_OUT) {
+		if (clip->lastSelectedParamID == CC_NUMBER_NONE) {
+			strcpy(parameterName, deluge::l10n::get(deluge::l10n::String::STRING_FOR_NO_PARAM));
+		}
+		else if (clip->lastSelectedParamID == CC_NUMBER_PITCH_BEND) {
+			strcpy(parameterName, deluge::l10n::get(deluge::l10n::String::STRING_FOR_PITCH_BEND));
+		}
+		else if (clip->lastSelectedParamID == CC_NUMBER_AFTERTOUCH) {
+			strcpy(parameterName, deluge::l10n::get(deluge::l10n::String::STRING_FOR_CHANNEL_PRESSURE));
+		}
+		else {
+			parameterName[0] = 'C';
+			parameterName[1] = 'C';
+			if (display->haveOLED()) {
+				parameterName[2] = ' ';
+				intToString(clip->lastSelectedParamID, &parameterName[3]);
+			}
+			else {
+				char* numberStartPos;
+				if (clip->lastSelectedParamID < 10) {
+					parameterName[2] = ' ';
+					numberStartPos = parameterName + 3;
+				}
+				else {
+					numberStartPos = (clip->lastSelectedParamID < 100) ? (parameterName + 2) : (parameterName + 1);
+				}
+				intToString(clip->lastSelectedParamID, numberStartPos);
+			}
+		}
+	}
+}
+
```

Comment:
updated function for displaying automation when playback is enabled (called from ui_timer_manager). Also used internally in the automation instrument clip view for updating the display and led indicators.

Reply:
Can this get a doc comment explanation too?